src/engraving/libmscore/edit.cpp

```diff
@@ -6054,6 +6051,18 @@ void Score::undoRemoveElement(EngravingItem* element)
     for (EngravingObject* ee : element->linkList()) {
         EngravingItem* e = static_cast<EngravingItem*>(ee);
         undo(new RemoveElement(e));
+
+        if (e->isMeasureRepeat()) {
+            const MeasureRepeat* repeat = toMeasureRepeat(e);
+            Measure* measure = repeat->firstMeasureOfGroup();
+            int staffIdx = repeat->staffIdx();
+
+            for (int i = 0; i < repeat->numMeasures(); ++i) {
+                undoChangeMeasureRepeatCount(measure, 0, staffIdx);
```

Comment:
We're calling this piece of code for all linked elements (`for (EngravingObject* ee : element->linkList())`), but the `undoChangeMeasureRepeatCount` method also does its job for all linked measures (`for (Staff* st : staff(staffIdx)->staffList())`). Isn't that redundant?